packages/synapse-interface/components/Portfolio/PortfolioContent.tsx

```diff
@@ -0,0 +1,176 @@
+import React, { useEffect } from 'react'
+import { Address } from 'wagmi'
+import {
+  NetworkTokenBalancesAndAllowances,
+  FetchState,
+} from '@/utils/hooks/usePortfolioBalances'
+import {
+  SingleNetworkPortfolio,
+  PortfolioAssetHeader,
+} from './SingleNetworkPortfolio'
+import { ConnectWalletButton } from './ConnectWalletButton'
+
+type PortfolioContentProps = {
+  connectedAddress: Address | string
+  connectedChainId: number
+  selectedFromChainId: number
+  networkPortfolioWithBalances: NetworkTokenBalancesAndAllowances
+  fetchPortfolioBalancesCallback: () => Promise<void>
+  fetchState: FetchState
+  bridgeTxHashes: string[]
+}
+
+export const PortfolioContent = ({
+  connectedAddress,
+  connectedChainId,
+  selectedFromChainId,
+  networkPortfolioWithBalances,
+  fetchPortfolioBalancesCallback,
+  fetchState,
+  bridgeTxHashes,
+}: PortfolioContentProps) => {
+  const { currentNetwork, remainingNetworks } = getCurrentNetworkPortfolio(
+    selectedFromChainId,
+    networkPortfolioWithBalances
+  )
+
+  useEffect(() => {
+    const txExists = bridgeTxHashes && bridgeTxHashes.length > 0
+    if (txExists) {
+      fetchPortfolioBalancesCallback()
+    }
+  }, [bridgeTxHashes])
+
+  return (
+    <div data-test-id="portfolio-content">
+      {currentNetwork && connectedChainId && selectedFromChainId && (
+        <SingleNetworkPortfolio
+          portfolioChainId={selectedFromChainId}
+          connectedChainId={connectedChainId}
+          selectedFromChainId={selectedFromChainId}
+          portfolioTokens={currentNetwork[selectedFromChainId]}
+          initializeExpanded={true}
+          fetchPortfolioBalancesCallback={fetchPortfolioBalancesCallback}
+          fetchState={fetchState}
+        />
+      )}
+      {connectedAddress ? (
+        fetchState === FetchState.LOADING ? (
+          <LoadingPortfolioContent />
+        ) : (
+          Object.keys(remainingNetworks).map(
+            (chainId: string, index: number) => {
+              const tokens = remainingNetworks[chainId]
+              return (
+                <SingleNetworkPortfolio
+                  portfolioChainId={Number(chainId)}
+                  connectedChainId={connectedChainId}
+                  selectedFromChainId={selectedFromChainId}
+                  portfolioTokens={tokens}
+                  initializeExpanded={false}
+                  fetchPortfolioBalancesCallback={
+                    fetchPortfolioBalancesCallback
+                  }
+                  fetchState={fetchState}
+                />
+              )
+            }
+          )
+        )
+      ) : (
+        <React.Fragment>
+          <PortfolioAssetHeader />
+          <UnconnectedPortfolioContent />
+        </React.Fragment>
+      )}
+    </div>
+  )
+}
+
+function getCurrentNetworkPortfolio(
```

Comment:
The getCurrentNetworkPortfolio helper could be improved by using filter instead of deleting:

```
function getCurrentNetworkPortfolio(currentChainId, networks) {

  const currentNetwork = networks[currentChainId]  

  const remainingNetworks = Object.keys(networks)
    .filter(key => key !== currentChainId)
    .map(key => ({[key]: networks[key]}))

  return { currentNetwork, remainingNetworks }

}
```

Reply:
In order to satisfy the conditions of `currentNetwork` and `remainingNetworks` being objects with key:array key-value pairs (in order to properly pass network:token data to the Portfolio component) , we would have to re-create the object as such: 

```
function getCurrentNetworkPortfolio(
  currentChainId: number,
  networks: NetworkTokenBalancesAndAllowances
): {
  currentNetwork: NetworkTokenBalancesAndAllowances
  remainingNetworks: NetworkTokenBalancesAndAllowances
} {
  const currentNetwork: NetworkTokenBalancesAndAllowances = {
    [currentChainId]: networks[currentChainId],
  }

  const remainingNetworks: NetworkTokenBalancesAndAllowances = Object.keys(
    networks
  )
    .filter((key: string) => Number(key) !== currentChainId)
    .reduce((result: NetworkTokenBalancesAndAllowances, key: string) => {
      result[Number(key)] = networks[key]
      return result
    }, {})

  return { currentNetwork, remainingNetworks }
}
```

Given the additional overhead using methods to iterate of keys of an object, think it would be more efficient to instead delete the key-value pair directly from the `remainingNetworks` object.